src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/ElinkUserServiceImpl.java

```diff
@@ -207,6 +204,23 @@ private ResponseEntity<Object> refreshUserProfileBasedOnCcdServiceNames(String c
         }
     }
 
+    private Page<UserProfile> fetchUserProfiles(PageRequest pageRequest,
+                                                Map<String, String> ccdServiceNameToCodeMapping,
+                                                List<String> ticketCode) {
+        Page<UserProfile> userProfilePage;
+        if (hasSpecialTribunalCic(ccdServiceNameToCodeMapping.keySet())) {
+            userProfilePage = userProfileRepository.fetchUserProfileByTicketCodes(ticketCode, pageRequest);
+        } else {
+            userProfilePage = userProfileRepository.fetchUserProfileByServiceNames(
+                    ccdServiceNameToCodeMapping.keySet(), ticketCode, pageRequest);
+        }
+        return userProfilePage;
+    }
+
+    private boolean hasSpecialTribunalCic(Set<String> serviceCodes) {
+        return serviceCodes.stream().anyMatch(serviceCode -> "BBA2".equalsIgnoreCase(serviceCode));
```

Comment:
minor: keySet() from 215 guarantees an initialised set, hence contains() method of set available, sufficient and preferred to stream processing